### ISSUE TITLE
Main CUL accessibility: validation errors

### ISSUE THREAD

**Issue by mhk33** [cul-it/uls#234]
Report specifically mentions <p> nested in <span>. I've emailed Sean Gnau to see how much needs to be validated, as there are many validation errors.

**Comment by mhk33** [comment]
Email thread with Sean Gnau about how to handle validation errors stemming from core and contributed modules:

 
 
From: Sean M Gnau <sg667@cornell.edu>
Date: Thursday, June 6, 2019 at 10:48 AM
To: Melissa Wallace <mhk33@cornell.edu>
Subject: Re: HTML validation
 
Hi Melissa,
 
That is honestly a tough one to answer definitively.  Ideally, all validation errors should be resolved whenever possible.  Examples of errors that really need to be resolved are ones where you have issues with ARIA, or repeating IDs, or you have a closing tag but no opening tag.  In cases where a module is generating these issues, we tend to have to rely on using Javascript to sort of “hack” it into compliance by removing attributes and/or adding the correct ones.
 
Errors related to invalid attributes should be cleaned up when possible, but ultimately most browsers are able to safely ignore invalid or deprecated attributes.  I would not go out of your way to resolve these unless there is a very simple way to edit the template files without compromising your ability to upgrade.
 
Hopefully this helps answer your questions
 
Regards,
Sean
 
From: Melissa Wallace <mhk33@cornell.edu>
Date: Thursday, June 6, 2019 at 9:37 AM
To: Sean M Gnau <sg667@cornell.edu>
Subject: HTML validation
 
Hi Sean,
 
I had a question about how far we go with HTML validation in SiteImprove. We have a number of errors for markup that come out of various Drupal modules. Do we need to get into the module code and fix it? For example, several modules put the type attribute on the script tag and apparently this is not valid. Also, there is markup generated by Drupal core that is not valid (e.g. <article>) and I don’t know if we need to bother to override it. We’d be diverging from Drupal core and modules, which would make it more difficult to upgrade later.
 
I guess I’m wondering if we have to get to 0 errors in validation, or if we just try to get to as close to error-free as we can. According to SiteImprove, our score is 95.2% and HTML validation would give us another .80 points.
 
Thanks for your help,
Melissa

**Comment by manolobevia** [comment]
@cappadona check Sean answer about HTML validation.

**Comment by mhk33** [comment]
These are the errors and warnings that are left for HTML validation:

(1) **Error coming from Antibot module, for all pages that have forms:**
Error: Element style not allowed as child of element noscript in this context. (Suppressing further errors from this subtree.)
From line 174, column 1; to line 174, column 7
noscript>↩<style>form.a

--> I've chosen not to try to fix this; will see what CIT says.

(2) **Warnings:**
<script type="text/javascript">
The type attribute is unnecessary for JavaScript resources.

--> This occurs 3x on all pages. The script tags are output by core, SiteImprove and something that references "beaconError."

(3) **44 errors on the path: library.cornell.edu/saml_login** --> This page is not presented to the user, it redirects to CUWebAuth or the user page if already logged in. Not sure if anything can be done with this.

Will see what CIT says about these.

CSS validation
https://use.fontawesome.com/releases/v5.1.0/css/all.css shows one error in SiteImprove, but when you click on the validation results, it says the file has no errors.

**Comment by manolobevia** [comment]
@mhk33 I have issues with the type attribute for <script> tag, but I think if I add a WP filter I will be able to get it off that. I ask myself how this will impact on the user? Also, I have so many 3rd party css files (from plugins) with errors on siteimprove, which I won't do anything. If I fix it, next time I update these plugins it will overwrite my changes.

**Comment by mhk33** [comment]
I suspect the script type attribute is not a big deal. SiteImprove actually shows a nice check mark and no errors on the pages that just have this warning.

I'm hopeful that we won't need to somehow fix any module/plug-in errors since they would likely get overwritten by updates. Sean seemed to say that he understood this.
